tests/integrations/test_db_integrations_sql.py

```diff
@@ -260,3 +409,23 @@ def test_db_system(self):
     @bug(library="python", reason="bug on pyodbc driver?")
     def test_db_user(self):
         super().test_db_user()
+
+    def test_obfuscate_query(self):
+        """ All queries come out obfuscated from agent """
+        for db_operation, request in self.requests[self.db_service].items():
+            span = self._get_sql_span_for_request(request)
+            # We launch all queries with two parameters (from weblog)
+            # Insert and procedure:These operations also receive two parameters, but are obfuscated as only one.
+            if db_operation in ["insert"] or (db_operation in ["procedure"] and context.library.library != "nodejs"):
```

Comment:
This if/else sequence is a little bit hard to read, and we don't understand why it's different on node. What about this : 

```python
if db_operation == "insert":
   expected_obfuscation_count = 1
elif db_operation == "procedure":
   # explanation why we expect only 0 obfuscation in node
   expected_obfuscation_count = 0 context.library.library == "nodejs" else 1
else:
  expected_obfuscation_count = 2

observed_obfuscation_count = span["meta"]["sql.query"].count("?")
assert (
    observed_obfuscation_count == expected_obfuscation_count
    ), f"The mssql query is not properly obfuscated for operation {db_operation}, expecting {expected_obfuscation_count} obfuscation(s), found {observed_obfuscation_count}:\n {span['meta']['sql.query']}"
```